time/src/duration.rs

```diff
@@ -41,7 +41,8 @@ type Nanoseconds =
 pub struct Duration {
     /// Number of whole seconds.
     seconds: i64,
-    /// Number of nanoseconds within the second. The sign always matches the `seconds` field.
+    /// Number of nanoseconds within the second. The sign always matches the `seconds` field,
+    /// except when `seconds` is 0.
```

Comment:
Also, even if you take your mentioned definiton on the face value, this is still not true, because a specific sign does not really match lack of sign.

Reply:
It's not a Rust thing so much as a number thing. However, Rust does have `is_positive` and `is_negative` methods for signed integers, both of which return false for zero. I'd be open to an "if any" type of wording, which would arguably be more correct. The wording as changed, though, I don't quite like.

Reply:
I believe this comment might have lead to the error that I'm fixing here in the first place - making the implementer think that they can just copy the `seconds`'s field sign and be done.

As you said, the sign of 0 is a "number thing" and it's completely arbitrary whether 0 is defined to have no sign, to have both positive and negative sign, or something yet else (e.g. distinct negative 0 and positive 0 that some arithmetics have).

Therefore, I'd really prefer to leave out any implicitly assumed defintion from here (that is required to understand what would "if any" mean), and just be specific that 0 is specifically excluded from the sign match.

Reply:
Given that I'm the one that implemented both, that's certainly not the case (I naturally know the many invariants present in the crate). This is a bug that wouldn't have been prevented by an addition to documentation here. I've probably never checked that documentation a single time after it was written, as I've internalized the invariant. Unfortunately that doesn't prevent bugs, as I am human and sometimes forget things. I would just as easily forget to check documentation, so that wouldn't help me out there.

The only time I'm aware of zero being considered anything other than unsigned is in number theory (where it can be either or both positive and negative), calculus (where it's zero _from a certain direction_), and floating point numbers (where it can be signed outright). To say the obvious, none of those are in any way relevant here; the sign of an `i64` is unambiguous in this context. Positive if greater than zero, negative if less than, and nonexistent if equal.

Reply:
I must say I don't understand your position. My comment makes the most possible explicit statement without needing any assumptions.

Your "if any" proposal requires the reader to understand that neither number theory, nor calculus, nor signed 0 arithmetics like floating point numbers have are relevant here, and in addition understand that "no sign present" is implied by both `.is_positive()` and `.is_negative()` returning false (because the Rust docs don't say anywhere explicitly what sign presence is/is not). That is a much bigger mental effort, even if the incorrect doc was not a cause of this particular bug.

Reply:
Reverted that change.